src/Service/Startup.cs

```diff
@@ -86,101 +87,38 @@ public void ConfigureServices(IServiceCollection services)
-            {
-                RuntimeConfigProvider configProvider = serviceProvider.GetRequiredService<RuntimeConfigProvider>();
-                RuntimeConfig runtimeConfig = configProvider.GetConfig();
-
-                return runtimeConfig.DataSource.DatabaseType switch
-                {
-                    DatabaseType.CosmosDB_NoSQL => null!,
-                    DatabaseType.MSSQL => ActivatorUtilities.GetServiceOrCreateInstance<MsSqlQueryBuilder>(serviceProvider),
-                    DatabaseType.PostgreSQL => ActivatorUtilities.GetServiceOrCreateInstance<PostgresQueryBuilder>(serviceProvider),
-                    DatabaseType.MySQL => ActivatorUtilities.GetServiceOrCreateInstance<MySqlQueryBuilder>(serviceProvider),
-                    _ => throw new NotSupportedException(runtimeConfig.DataSource.DatabaseTypeNotSupportedMessage),
-                };
-            });
 
             services.AddSingleton<ILogger<ISqlMetadataProvider>>(implementationFactory: (serviceProvider) =>
             {
                 ILoggerFactory? loggerFactory = CreateLoggerFactoryForHostedAndNonHostedScenario(serviceProvider);
                 return loggerFactory.CreateLogger<ISqlMetadataProvider>();
             });
 
-            services.AddSingleton<ISqlMetadataProvider>(implementationFactory: (serviceProvider) =>
+            // Below are the factory registrations that will enable multiple databases scenario.
+            // within these factories the various instances will be created based on the database type and datasourceName.
+            services.AddSingleton<IQueryManagerFactory>(implementationFactory: (serviceProvider) =>
             {
-                RuntimeConfigProvider configProvider = serviceProvider.GetRequiredService<RuntimeConfigProvider>();
-                RuntimeConfig runtimeConfig = configProvider.GetConfig();
+                return ActivatorUtilities.GetServiceOrCreateInstance<QueryManagerFactory>(serviceProvider);
+            });
 
-                return runtimeConfig.DataSource.DatabaseType switch
-                {
-                    DatabaseType.CosmosDB_NoSQL => ActivatorUtilities.GetServiceOrCreateInstance<CosmosSqlMetadataProvider>(serviceProvider),
-                    DatabaseType.MSSQL => ActivatorUtilities.GetServiceOrCreateInstance<MsSqlMetadataProvider>(serviceProvider),
-                    DatabaseType.PostgreSQL => ActivatorUtilities.GetServiceOrCreateInstance<PostgreSqlMetadataProvider>(serviceProvider),
-                    DatabaseType.MySQL => ActivatorUtilities.GetServiceOrCreateInstance<MySqlMetadataProvider>(serviceProvider),
-                    _ => throw new NotSupportedException(runtimeConfig.DataSource.DatabaseTypeNotSupportedMessage),
-                };
+            services.AddSingleton<IQueryEngineFactory>(implementationFactory: (serviceProvider) =>
+            {
+                return ActivatorUtilities.GetServiceOrCreateInstance<QueryEngineFactory>(serviceProvider);
```

Comment:
Can you add a code comment that describes why the next few uses of `AddSingleton<T>()` require usage of `implementationFactory` param with activatorUtilities?

e.g. why not just use
```csharp
services.AddSingleton<IQueryManagerFactory, QueryManagerFactory>()`;
```

Is it because the above usage instantiates immediately, which would fail in a SWA scenario where config is not immediately available?

Reply:
No longer using implementationFactory here, directly using as mentioned in example.